docs/adr/00xx-moving-business-logic-out-of-controllers.md

```diff
@@ -0,0 +1,124 @@
+  that has embraced the collection of Trailblazer utilities.
+- Publishing API command pattern: This is an effective pattern at moving logic
+  out of controllers, it however has produced usage problems where the base
+  class is coupled to particular domain logic.
+
+## Decision
+
+Controller actions can use a dedicated class to perform the business logic of
+a request. Logic in an action is to be focused on producing the appropriate
+HTTP response. To meet this requirement it is expected that classes shall
+exist for actions that mutate application state (typically POST/PUT/DELETE
+requests) and may exist for actions which don’t mutate state (typically GET)
+depending on the request complexity.
+
+Content Publisher makes use of the [Interactor gem][interactor-gem] as the
+means to implement these classes. It was chosen over other community
+approaches due to it’s simplicity and flexibility. It was decided to use
+a community provided framework rather that writing our own, this is
+because:
+
+- a bespoke problem is not being solved, so it shouldn't require a bespoke
+  solution;
+- authoring a new pattern is something that requires negotiation,
+  documentation, maintenance and iteration;
+- It’s harder to [bikeshed][] on decisions made externally.
+
+The Interactor gem is relativity simple in comparison to similar gems. This
+offers the advantage that, were we to discover problems with the gem or a
+change in our needs, it would be realistic for us to adapt this to a self
+built or different community approach.
+
+Interactor classes are created with a coupling to a particular controller
+action. Interactors are stored in an app/interactors directory. Within here
+there are directories created for each controller to store the interactors for
+each action, in a pattern similar to views. Each interactor class name is
+suffixed with "Interactor", to make their purpose clear. As an example,
+given a create action on a DocumentsController there is a corresponding
+interactor to create a document, Document::CreateInteractor, which is stored
+in app/interactors/document.
+
+## Status
+
+Accepted
+
+## Consequences
+
+Many of the controller actions in Content Publisher have been refactored to
+make use of the interactor pattern. This has resulted in an increase in the
+number of classes of the application while the size and complexity of
+controller classes have decreased.
+
+Conventions have been established on a consistent approach to using the
+Interactor gem. These have focused on:
+
+- [initialising controllers with request parameters and a user][initialize-example];
+- [authoring an interactor's `call` method as a procedure of descriptive
+  methods][call-method-example];
+- [using delegation to reduce repetitive references to the `context`
+  variable][delegation-example];
+- [setting variables on context to be used to determine HTTP
+  response][response-example],
+  rather than using context.success / context.failure methods, this allows the
```

Comment:
Minor, but code syntax might help a bit here. Also, I think it's conventional to write `Context#failure`, since `Context.failure` might be confused as referring to a class method (under this same convention). Pretty sure I read that in a book somewhere!

Reply:
Thanks - yeah that's a common convention. I was referencing the instance with the lower case context but your suggestion looks an improvement.